examples/binding-patterns-in-match-statement/binding_patterns_in_match_statement.bal

```diff
@@ -0,0 +1,25 @@
+import ballerina/io;
+
+type Pair record {
+    int x;
+    int y;
+};
+
+function matchFn(Pair pair) {
+    match pair {
+        // The binding pattern below also has a rest binding pattern to capture the additional fields
```

Comment:
Isn't this BBE meant to be covering binding patterns in a match statement in general? Right now it's kinda emphasizing the rest binding pattern. Let's improve the comments and add more examples if required. Looks like we're missing some stuff here from the slides too.